main/streams/plain_wrapper.c

```diff
@@ -143,7 +143,7 @@ typedef struct {
 } php_stdio_stream_data;
 #define PHP_STDIOP_GET_FD(anfd, data)	anfd = (data)->file ? fileno((data)->file) : (data)->fd
 
-static int do_fstat(php_stdio_stream_data *d, int force)
+static zend_bool do_fstat(php_stdio_stream_data *d, int force)
```

Comment:
Oo. I read it as r = d->cached_fstat == 0

I'll change it back to int, and change the checking.